temp/OnnxTransformTempSolution/Microsoft.ML.Transforms.Onnx/OnnxUtils.cs

```diff
@@ -0,0 +1,209 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Scoring;
+using OnnxShape = System.Collections.Generic.List<long>;
+
+namespace Microsoft.ML.Transforms.Onnx
+{
+    /// <summary>
+    /// OnnxModelInfo contains the data that we should get from 
+    /// Sonoma API once that functionality is added.
+    /// </summary>
+    public sealed class OnnxModelInfo
+    {
+        public OnnxNodeInfo[] InputsInfo;
+        public OnnxNodeInfo[] OutputsInfo;
+    }
+
+    /// <summary>
+    /// OnnxNodeInfo contains all the information for a given node (e.g. inputs/outputs)
+    /// of an Onnx model.
+    /// </summary>
+    public class OnnxNodeInfo
+    {
+        public string Name;
+        public OnnxShape Shape;
+        public Type DataType = typeof(System.Single);
+    }
+
+    /// <summary>
+    /// OnnxModel is a facad for ModelManager. ModelManager is provided by Sonoma API, 
+    /// and it has a lot of functionality (multiple models, multiple versions) that are not
+    /// needed by Onnx transform, which only needs a single model. This facad simplifies the
+    /// usage of onnx model. 
+    /// </summary>
+    internal sealed class OnnxModel
+    {
+        private static readonly int IgnoredVersion = int.MaxValue;
+        private ModelManager _modelManager;
+        private string _modelName;
+        private readonly List<string> _inputNames;
+        private readonly List<string> _outputNames;
+
+        public OnnxModel(string modelFile, OnnxModelInfo modelInfo)
+        {
+            // Load the onnx model
+            var modelFileInfo = new FileInfo(modelFile);
+            _modelName = Path.GetFileNameWithoutExtension(modelFileInfo.Name);
+            _modelManager = new ModelManager(modelFileInfo.Directory.FullName, true);
+            _modelManager.InitModel(_modelName, IgnoredVersion);
+
+            _inputNames = modelInfo.InputsInfo.Select(i => i.Name).ToList();
+            _outputNames = modelInfo.OutputsInfo.Select(i => i.Name).ToList();
+        }
+
+        public OnnxModel(byte[] modelBytes)
+        {
+            throw new NotImplementedException("Need an API to serialize/deserialize onnx models to byte arrays!");
```

Comment:
@Jignesh - we need API to serialize/deserialize onnx models to byte arrays for saving and loading ml.net models. Does it exists in sonoma?

Reply:
@shmoradims ,this method needs to be added.  We can add as follows.

byte[]^ GetModelAsBytes((System::String^ modelName, int32_t version);

Reply:
That would work for model->bytes. For bytes->model, I'd need to save the bytes to a temp file and the use the ctor(string filepath) and cleanup. I can do that, but it would make sense for the sonoma API to handle that directly with a ctor(byte[]).

Reply:
Can we do the following?

1> To unblock demo, use the approach mentioned above (using temp file), similar to Python Trainer transform. This  should be relatively quick stopgap fix.
2> In Sonoma, after we remove dependency on manifest file + model_export.exe (will take ~1 week or longer), we can add the API to serialize model to/from byte[] , and  upgrade ml.net onnx transform accordingly.

The cleaner API (in Sonoma) requires quite a bit of refactoring -- the steps above would unblock the demo milestone, which we get the APIs in place for the Sonoma/ML.net release milestone.